packages/web/components/bridge/immersive/crypto-fiat-input.tsx

```diff
@@ -0,0 +1,226 @@
+            />
+          ) : (
+            <InputBox
+              // when the font size changes, we need to prompt the autosize input to re-mount
+              // see: https://github.com/JedWatson/react-input-autosize?tab=readme-ov-file#changing-the-styles-at-runtime
+              key={cryptoInputFontSize}
+              inputRef={inputRef}
+              className={classNames(
+                "border-none bg-transparent font-bold",
+                cryptoInputFontSize
+              )}
+              classes={{
+                label: "!block",
+                input: classNames("!p-0", {
+                  "text-rust-300": isInsufficientBal || isInsufficientFee,
+                }),
+                trailingSymbol: classNames(
+                  "ml-1 align-middle text-osmoverse-500",
+                  {
+                    "text-rust-300": isInsufficientBal || isInsufficientFee,
+                  }
+                ),
+              }}
+              onBlur={() => setIsInputFocused(false)}
+              onFocus={() => setIsInputFocused(true)}
+              currentValue={cryptoInputRaw}
+              onInput={onInput("crypto")}
+              trailingSymbol={cryptoAmountPretty.denom}
+              isAutosize
+            />
+          )}
+        </div>
+        <button
+          onClick={() => {
+            // TODO: Subtract fees from the amount
+            onInput("crypto")(
+              trimPlaceholderZeros(asset.amount.toDec().toString())
+            );
+          }}
+          className="body2 w-14 shrink-0 transform rounded-5xl border border-osmoverse-700 py-2 px-3 text-wosmongton-200 transition duration-200 hover:border-osmoverse-850 hover:bg-osmoverse-850 hover:text-white-full disabled:opacity-80"
+        >
+          {t("transfer.max")}
+        </button>
+      </div>
+
+      <button
+        className="body1 flex items-center gap-2 text-center text-wosmongton-200"
+        onClick={() => {
+          setInputUnit(currentUnit === "fiat" ? "crypto" : "fiat");
+        }}
+      >
+        <span>
+          {currentUnit === "fiat" ? (
+            <>
+              {trimPlaceholderZeros(
+                cryptoAmountPretty?.toDec().toString(2) ?? "0"
+              )}{" "}
+              {cryptoAmountPretty.denom}
+            </>
+          ) : (
+            fiatAmountPretty.maxDecimals(2).toString()
+          )}
+        </span>
+        <span>
+          <Icon
+            id="switch"
+            className="text-wosmongton-200"
+            width={16}
+            height={16}
+          />
+        </span>
+      </button>
+    </div>
+  );
+};
+
+function calcTextSizeClass(numChars: number): string {
+  if (numChars <= 8) {
+    return "text-4xl";
+  } else if (numChars <= 10) {
+    return "text-3xl";
+  } else if (numChars <= 12) {
+    return "text-2xl";
+  } else if (numChars <= 18) {
+    return "text-xl";
+  } else if (numChars <= 24) {
+    return "text-lg";
+  }
+
+  return "text-md";
+}
```

Comment:
I wonder how scalable this is. Nass previously created a [code sandbox](https://codepen.io/nassdonald/pen/RwmMMzK) that scales the input based on the content. I believe that solution would be better than hardcoding the text classes and would also be smoother during size transitions. Additionally, I'm a bit concerned about the autosize library. It seems we need many workarounds to get it to function properly

Reply:
I think we could use GPT to convert it to a react component and modify it promptly for this case

Reply:
I agree that works really smoothly, but for sake of time can we run with this then adapt the codepen solution later if we decide to?

Reply:
agree with @jonator

Reply:
This existing solution, re-using our already-imported react-autosize-input package, already took several hours to get working smoothly in a way that matches the Figma designs.